_templates/l10n.php

```diff
@@ -35,334 +37,368 @@ function list_langs() {
+
+        if (!$this->is_lang($lang)) {
+            return false;
+        }
 
-function load_translations($index, $lang) {
-    if (!is_lang($lang)) {
-        return false;
+        $langFile = $this->lang_dir($lang).'/'.$index.'.json';
+        if (!file_exists($langFile)) {
+            return false;
+        }
+
+        $json = file_get_contents($langFile);
+        return json_decode($json, true);
     }
 
-    $langFile = lang_dir($lang).'/'.$index.'.json';
-    if (!file_exists($langFile)) {
-        return false;
+    protected function load_domain($domain) {
+        $this->translations[$domain] = $this->load_translations($domain);
     }
 
-    $json = file_get_contents($langFile);
-    return json_decode($json, true);
-}
+    public function set_domain($domain) {
+        if (ob_get_level()) {
+            ob_flush(); // Flush output buffer
+        }
 
-$l10nDomain = null;
-$translations = array();
-function set_l10n_domain($domain) {
-    global $lang, $l10nDomain, $translations;
+        $this->domain = $domain;
 
-    if (ob_get_level()) {
-        ob_flush(); // Flush output buffer
+        if (!isset($this->translations[$domain])) {
+            $this->load_domain($domain);
+        }
     }
 
-    $l10nDomain = $domain;
-
-    if (!isset($translations[$domain])) {
-        $translations[$domain] = load_translations($domain, $lang);
-    }
-}
+    /**
+     * Translate a string. Returns the original string if no translation was found.
+     */
+    public function translate($id, $domain = null, $string = null) {
+        if (empty($domain)) {
+            $domain = $this->domain;
+        }
+        if (empty($string)) {
+            $string = $id;
+        }
+        if (!isset($this->translations[$domain])) {
+            $this->load_domain($domain);
+        }
 
-/**
- * Translate a string. Returns the original string if no translation was found.
- */
-function translate($id, $domain, $string) {
-    global $translations;
-
-    if (isset($translations[$domain][$id]) &&
-        is_string($translations[$domain][$id])) {
-        return $translations[$domain][$id];
-    } else {
-        return $string;
+        if (isset($this->translations[$domain][$id]) &&
+            is_string($this->translations[$domain][$id])) {
+            return $this->translations[$domain][$id];
+        } else {
+            return $string;
+        }
     }
-}
 
-/**
- * Translate a HTML string. This includes a minimalist XML parser.
- * Can be provided a $translate callback to override default behaviour.
- * (Useful to extract strings from a file for instance.)
- */
-function translate_html($input, $translate = 'translate') {
-    global $l10nDomain;
+    /**
+     * Translate a HTML string. This includes a minimalist XML parser.
+     * Can be provided a $translate callback to override default behaviour.
+     * (Useful to extract strings from a file for instance.)
+     */
+    public function translate_html($input, $translate = null) {
```

Comment:
This method is doing more than the name suggests. 

It should just translate a string, not parse XML, doing security, etc. You could split it in separate methods and use a proper [XML parser](http://php.net/manual/en/book.xml.php) instead of inventing your own.